src/sync.d

```diff
@@ -6815,7 +6829,7 @@ class SyncEngine {
 			if (!itemDB.selectByPath(oldPath, appConfig.defaultDriveId, oldItem)) {
 				// The old path|item is not synced with the database, upload as a new file
 				addLogEntry("Moved local item was not in-sync with local databse - uploading as new item");
-				uploadNewFile(newPath);
+				scanLocalFilesystemPathForNewData(newPath);
```

Comment:
Why scan the entire filesystem, rather than pushing and using the uploadNewFile function directly?

Reply:
The `scanLocalFilesystemPathForNewData` function can handle file uploads as well. In this new version, for cases like `mkdir new_folder && mv new_folder new_loc`, new_folder will not trigger a redundant online creation (as local dir is gone). Consequently, it will not be in sync and will require a filesystem scan.